core/src/main/scala/kafka/tools/MirrorMaker.scala

```diff
@@ -434,6 +434,10 @@ object MirrorMaker extends Logging with KafkaMetricsGroup {
               records.foreach(producer.send)
               maybeFlushAndCommitOffsets()
             }
+            if (!mirrorMakerConsumer.hasData && !shuttingDown) {
```

Comment:
Maybe we need to add check `!exitingOnSendFailure` as well? Otherwise we may throw IllegalStateException when there is a send failure.